x/asset/keeper/keeper.go

```diff
@@ -35,33 +35,32 @@ func NewKeeper(
 	ps paramtypes.Subspace,
 	bankKeeper types.BankKeeper,
 	ak types.AccountKeeper,
-	privileges ...types.PrivilegeI,
 ) *Keeper {
 	// set KeyTable if it has not already been set
 	if !ps.HasKeyTable() {
 		ps = ps.WithKeyTable(types.ParamKeyTable())
 	}
 
-	privilegesMap := map[string]types.PrivilegeI{}
-	for _, priv := range privileges {
-		if _, ok := privilegesMap[priv.Name()]; ok {
-			continue
-		}
-
-		privilegesMap[priv.Name()] = priv
-		// regiester the privilege's interfaces
-		priv.RegisterInterfaces()
+	return &Keeper{
+		cdc:              cdc,
+		storeKey:         storeKey,
+		memKey:           memKey,
+		paramstore:       ps,
+		bankKeeper:       bankKeeper,
+		ak:               ak,
+		PrivilegeManager: map[string]types.PrivilegeI{},
 	}
+}
 
-	return &Keeper{
-		cdc:           cdc,
-		storeKey:      storeKey,
-		memKey:        memKey,
-		paramstore:    ps,
-		bankKeeper:    bankKeeper,
-		ak:            ak,
-		PrivilegesMap: privilegesMap,
+func (k Keeper) AddPrivilege(priv types.PrivilegeI) error {
```

Comment:
why this here, since privilege is set only during keeper initialization, we should not exposed an API to allow to add more privilege

Reply:
cause this is a clean way to input the privileges instead of a map. Also, We want to explicitly add each privilege by a separate API call (like when we add middleware to ibc transfer for example) instead of grouping them together in a map.